scripts/safe_template_linter.py

```diff
@@ -817,39 +868,117 @@ def _find_mako_expressions(self, mako_template):
 
         return expressions
 
-    def _find_balanced_end_curly(self, mako_template, start_index, num_open_curlies):
+    def _find_closing_char_index(self, start_delim, open_char, close_char, template, start_index, num_open_chars):
         """
-        Finds the end index of the Mako expression's ending curly brace.  Skips
-        any additional open/closed braces that are balanced inside.  Does not
-        take into consideration strings.
+        Finds the index of the closing char that matches the opening char.
+
+        For example, this could be used to find the end of a Mako expression,
+        where the open and close characters would be '{' and '}'.
 
         Arguments:
-            mako_template: The template text.
-            start_index: The start index of the Mako expression.
-            num_open_curlies: The current number of open expressions.
+            start_delim: If provided (e.g. '${' for Mako expressions), the
+                closing character must be found before the next start_delim.
+            open_char: The opening character to be matched (e.g '{')
+            close_char: The closing character to be matched (e.g '}')
+            template: The template to be searched.
+            start_index: The start index of the last open char.
+            num_open_chars: The current number of open chars.
 
         Returns:
-            The end index of the expression, or -1 if unparseable.
+            The index of the closing character, or -1 if unparseable.
         """
-        end_curly_index = mako_template.find('}', start_index)
-        if end_curly_index < 0:
-            # if we can't find an end_curly, let's just quit
-            return end_curly_index
+        end_char_index = template.find(close_char, start_index)
```

Comment:
Having `end_char_index` and `close_char` be named differently reads odd to me, I'd rather they both be named `end` or `close`